src/main/java/org/zalando/nakadi/service/FloodService.java

```diff
@@ -0,0 +1,177 @@
+package org.zalando.nakadi.service;
+
+import org.apache.commons.collections.map.HashedMap;
+import org.apache.curator.framework.recipes.cache.ChildData;
+import org.apache.curator.framework.recipes.cache.TreeCache;
+import org.apache.zookeeper.KeeperException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.zalando.nakadi.domain.EventType;
+import org.zalando.nakadi.exceptions.NakadiException;
+import org.zalando.nakadi.repository.db.EventTypeCache;
+import org.zalando.nakadi.repository.zookeeper.ZooKeeperHolder;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+@Component
+public class FloodService {
```

Comment:
> Got it. +1 on the basis this will get revisited.

Do you think it is really harmful to use Curator's TreeCache for a big number of read operations? 
We will write directly to zk in very rare cases when we really need to block/unblock a particular client.